### PR TITLE
refactor: slight refactor on the UPE available methods

### PR DESCRIPTION
**Pull request by frosso**
Fixes #

#### Changes proposed in this Pull Request

Ensures that when the UPE flag is disabled, only `card` is returned as an available payment method.

Some inconsistency has been created.
We needed to add additional available payment methods. But UPE was not enabled.
So we created the `get_upe_available_payment_methods` method with the `wcpay_upe_available_payment_methods` filter.
Then the UPE gateway got introduced, and inherited the method. All good.
But as work progressed, we forgot to _only_ return the additional payment methods when UPE is enabled.

Adding some consistency, so that there's less confusion if the flags are manually switched.

#### Testing instructions

- With "grouped settings" enabled
  - Just UI changes on the settings
- With "grouped settings" and "Enable UPE settings changes"
  - Adds "Payments accepted on checkout" section in the new settings: https://d.pr/i/TZKnu5
- With "grouped settings" and "Enable UPE checkout"
  - Just UI changes on the settings
  - Adds "Setup additional payment methods" task: https://d.pr/i/mt7n4b
- With "grouped settings" and "Add UPE additional payment methods"
  - Just UI changes on the settings - this is not a valid combination of flags we're going to release ("Add UPE additional payment methods" is technically dependent on "Enable UPE checkout")

Feel free to toggle between feature flags, things should be more consistent, now.

Having "grouped settings" and "Enable UPE checkout" is the only "fishy" one, because it adds the "Setup additional payment methods" task.
But we're going to launch with that at the moment, so I personally think it makes sense.

-------------------

- [ ] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/class-wc-payment-gateway-wcpay.php [hunk 1]
```diff
@@ -2167,11 +2167,8 @@ public function get_upe_enabled_payment_method_ids() {
 	 * @return string[]
 	 */
 	public function get_upe_available_payment_methods() {
-		return apply_filters(
```
Comment (frosso) [inline includes/class-wc-payment-gateway-wcpay.php:2170]: Before: we were allowing the dev tools to append additional payment methods when the UPE flag was disabled

File: includes/payment-methods/class-upe-payment-gateway.php [hunk 2]
```diff
@@ -527,4 +527,25 @@ public function payment_fields() {
...
+	 * @return string[]
+	 */
+	public function get_upe_available_payment_methods() {
+		$methods = parent::get_upe_available_payment_methods();
+
+		$methods[] = 'giropay';
+		$methods[] = 'sofort';
+
+		return array_values(
+			apply_filters(
```
Comment (frosso) [inline includes/payment-methods/class-upe-payment-gateway.php:545]: Now: we are allowing the dev tools to append additional payment methods only when the UPE flag is enabled

File: includes/payment-methods/class-upe-payment-gateway.php [hunk 3]
```diff
@@ -527,4 +527,25 @@ public function payment_fields() {
...
+	/**
+	 * Returns the list of available payment method types for UPE.
+	 * See https://stripe.com/docs/stripe-js/payment-element#web-create-payment-intent for a complete list.
+	 *
+	 * @return string[]
+	 */
+	public function get_upe_available_payment_methods() {
+		$methods = parent::get_upe_available_payment_methods();
+
+		$methods[] = 'giropay';
```
Comment (frosso) [inline includes/payment-methods/class-upe-payment-gateway.php:541]: Now: enabling `giropay` and `sofort` in UPE (other payment methods can be added by the dev tools)

**Comment by harriswong** [comment]
👀

**Review by harriswong — APPROVED** [review]
Makes sense that dev tool's `Add UPE additional payment methods` has no effect on `WC_Payment_Gateway_WCPay` class. And only has an effect on the UPE gateway class `UPE_Payment_Gateway`. 

👍  LGTM but would like to hear what @waclawjacek thinks as well.

**Review by waclawjacek — APPROVED** [review]
Awesome! 🚢

**Pull request merged**